### PR TITLE
bpo-22367: Update test_fcntl.py for spawn process mode

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue22367

### CODE REVIEW

**Comment by corona10** [comment]
@vstinner I've checked both Linux and mac works as expected on this test.
Please take a look

**Comment by corona10** [comment]
I add @zooba as the reviewer also. Since https://bugs.python.org/issue38790 was reported also.

**Review by zooba — APPROVED** [review]
This appears to fix the tests as far as Azure Pipelines is concerned.

I have no familiarity with what this is actually doing though, so I can't say whether the test is still valid. Get a second approval besides mine before merging.

**Pull request merged**
